traitsui/qt4/file_editor.py

```diff
@@ -131,17 +125,13 @@ def get_error_control(self):
 
     def _create_file_dialog(self):
         """Creates the correct type of file dialog."""
-        dlg = QtGui.QFileDialog(self.control)
-        dlg.selectFile(self._file_name.text())
-
-        if self.factory.dialog_style == "open":
-            dlg.setAcceptMode(dlg.AcceptOpen)
-        elif self.factory.dialog_style == "save":
-            dlg.setAcceptMode(dlg.AcceptSave)
-
-        if len(self.factory.filter) > 0:
-            dlg.setNameFilters(self.factory.filter)
-
+        wildcard = " ".join(self.factory.filter)
+        dlg = FileDialog(
+            parent=self.get_control_widget(),
+            default_path=self._file_name.text(),
+            action="save" if self.factory.dialog_style == "save as" else "open",
+            wildcard=wildcard,
+        )
```

Comment:
`_create_control` for the FileDialog seems slightly more involved than what was done here previously but I trust pyface is doing the reasonable things

Reply:
The default path handles the selection of the file; the action handles the accept mode.